contents/tutorials/astro-surveys.md

```diff
@@ -0,0 +1,628 @@
+import CustomSurvey from '../components/CustomSurvey.astro';
+---
+<Layout>
+	<html lang="en">
+		<head>
+			<meta charset="utf-8" />
+			<link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+			<meta name="viewport" content="width=device-width" />
+			<meta name="generator" content={Astro.generator} />
+			<title>Astro</title>
+		</head>
+		<body>
+			<h1>Astro Surveys</h1>
+			<CustomSurvey class="survey" />
+			
+			<script>
+				function toggleSurveyDisplay(show) {
+				  const surveyElement = document.querySelector('.survey');
+				  if (surveyElement) {
+				    surveyElement.style.display = show ? 'block' : 'none';
+				  }
+				}
+
+				function handleDismiss() {
+					toggleSurveyDisplay(false);
+				}
+
+				function handleSubmit(event) {
+					toggleSurveyDisplay(false);
+				}
+
+				if (typeof window !== 'undefined') {
+				  window.addEventListener('dismiss_survey', handleDismiss);
+				  window.addEventListener('submit_survey', handleSubmit);
+				}
+			</script>
+
+		</body>
+	</html>
+</Layout>
+```
+
+This shows a survey popup every time you visit your app's homepage.
+
+![Custom survey UI](../images/tutorials/astro-surveys/sample-survey-ui.png)
+
+#### 2. Fetch the survey from PostHog
+
+PostHog keeps track of all active surveys for a user. This is especially helpful if you set up [custom targeting options](/docs/surveys/creating-surveys#targeting). 
+
+To fetch the active surveys, we use `posthog.getActiveMatchingSurveys()`. This returns an array of survey objects that looks like this:
+
+```JSON
+[
+   {
+     "id": "018cfcd5-107e-0000-49a1-8e7c6b825947",
+     "name": "Net promoter score (NPS) API Survey",
+     "description": "",
+     "type": "api",
+     "linked_flag_key": null,
+     "targeting_flag_key": null,
+     "questions": [
+       {
+         "type": "rating",
+         "scale": 10,
+         "display": "number",
+         "question": "How likely are you to recommend us to a friend?",
+         "description": "",
+         "lowerBoundLabel": "Unlikely",
+         "upperBoundLabel": "Very likely"
+       }
+     ],
+     "conditions": null,
+     "start_date": "2024-01-12T08:41:20.614000Z",
+     "end_date": null
+   }
+]
+```
+
+To fetch this array and integrate it with your survey UI, update your `<script>` in `index.astro`:
+
+```astro file=index.astro
+---
+import Layout from '../layouts/Layout.astro';
+import CustomSurvey from '../components/CustomSurvey.astro';
+---
+<Layout>
+	<html lang="en">
+		<head>
+			// your existing code
+		</head>
+		<body>
+			// your existing code
+      			
+			<script>
+				let surveyID = '';
+				document.addEventListener('DOMContentLoaded', async () => {
+					window.posthog.getActiveMatchingSurveys((surveys) => {
+						if (surveys.length > 0) {
+							const survey = surveys[0];
+							document.getElementById('survey-title').textContent = survey.questions[0].question;
+							surveyID = survey.id;
+						}
+					});
+				});
```

Comment:
Maybe it is the GitHub styling, but would this be easier to read if it was only the change and we remove a lot of the `// your existing code` stuff?